using-kontena/nodes.md

```diff
@@ -47,15 +76,30 @@ The command that may be used for connecting to Kontena Node via SSH.
 $ kontena node ssh <NODE_ID>
 ```
 
+## Reset a Kontena Node Token
+
+The `kontena node reset-token` command can be used to replace a compromised node token, upgrade a node that was originally provisioned using a grid token, or revert a node to using a grid token:
+
+```
+$ kontena node reset-token <NODE_ID>
+```
+
+Resetting the token of an online Kontena Node will force the agent to disconnect. The agent will not be able to reconnect until the the node is reconfigured using the new `kontena node env` values.
+
+The `kontena node reset-token` command can also be used to upgrade a node provisioned using a grid token. The node must be reconfigured using the new `kontena node env` configuration before it will be able to reconnect. The `kontena node reset-token --clear-token` command can be used to revert back to the grid token.
+
+
 ## Remove a Kontena Node
 
-In order to remove a Kontena Node, it must be terminated first. Once terminated, it may be removed. Only `offline` Kontena Nodes may be removed.
+The `kontena node remove` command can be used to remove a node that is being decomissioned:
 
 ```
 $ kontena node remove <NODE_ID>
 ```
 
-If you are using the Kontena CLI built-in provision tool, you can terminate Kontena Node using the `kontena <provider> node terminate` command. Alternatively, power off / terminate the Kontena Node machine from any infrastructure you are using and wait for the machine to become offline before removing them.
+If the node was provisioned using the Kontena CLI built-in provisioning tool, you can terminate the host machine using the `kontena <provider> node terminate` command. This will also remove the Kontena Node. If the host machine has already been terminated, then the `kontena node remove` command can be used to forget the terminated node.
+
+If the node was provisioned with a node token, then the `kontena node remove` command can also be used to invalidate the node token, forcing the agent to disconnect if it is still connected. Nodes provisioned using grid tokens cannot be removed if they are still online, because the agent would simply reconnect and the node would quickly re-appear.
```

Comment:
> Remove Kontena Node - Is there additional value of having longer story?

There are now two kinds of nodes: those with grid tokens and those with node tokens. They behave differently in terms of `kontena node remove`.

The `kontena <provider> node terminate` is still the main preferred thing to use. Removing an online node provisioned using a node token will still leave the machine running. That implies something like you losing access to that node and wanting to kick it out of your grid ASAP.

Should we drop support for removing online nodes connected using node tokens, to simplify the docs? You can still use `kontena node reset-token` to kick it offline, and then remove it.